app/src/main/java/com/ably/tracking/demo/publisher/ably/log/LocationLogger.kt

```diff
@@ -0,0 +1,76 @@
+package com.ably.tracking.demo.publisher.ably.log
+
+import com.ably.tracking.Location
+import com.ably.tracking.LocationUpdate
+import com.ably.tracking.publisher.LocationHistoryData
+import com.google.gson.Gson
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+
+class LocationLogger(
+    private val fileWriter: LogFileWriter,
+    private val fileManager: FileManager,
+    private val gson: Gson
+) {
+
+    companion object {
+        private const val LOG_TIME_FORMATTER_PATTERN = "HH:mm:ss"
+        private const val FILE_NAME_FORMATTER_PATTERN = "dd.MM_HH:mm:ss"
+        private const val LOG_FILE_NAME_SUFFIX = "_location.log"
+        private const val HISTORY_FILE_NAME_SUFFIX = "_history.log"
+        private const val LOG_DIRECTORY = "riderLogs"
+        private const val SPACE = " "
+    }
+
+    private var sessionStart: Long = 0
+
+    private val logTimeFormatter by lazy {
+        SimpleDateFormat(LOG_TIME_FORMATTER_PATTERN, Locale.getDefault())
+    }
+
+    fun logLocationUpdate(locationUpdate: LocationUpdate) {
+        val location = locationUpdate.location
+        ensureLogFileExists(location)
+        logLocation(location)
+    }
+
+    private fun ensureLogFileExists(location: Location) {
+        if (!fileWriter.isReady) {
+            sessionStart = location.time
+            val fileName = sessionStart.toFileName() + LOG_FILE_NAME_SUFFIX
+
+            fileWriter.prepare(LOG_DIRECTORY, fileName)
+        }
+    }
+
+    private fun logLocation(location: Location) {
+        val date = Date(location.time)
+        fileWriter.writeLine(logTimeFormatter.format(date) + SPACE + location)
```

Comment:
You could use Kotlin's string formatting syntax to get rid of this `SPACE` constant :wink: But both versions are fine, so I'll let you decide :+1: 
```suggestion
        fileWriter.writeLine("${logTimeFormatter.format(date)} $location")
```

Reply:
I would prefer to stick to `SPACE` constant, I think it's easy to overlook space between `}` and `$`